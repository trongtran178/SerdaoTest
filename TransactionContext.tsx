import React, {createContext, useState, useContext} from 'react';
import {Beneficiary, Transaction} from './models';

type TransactionContextType = {
  transactions: Transaction[];
  addTransaction: (amount: string, account: Beneficiary) => void;
  balance: number;
};
const TransactionContext = createContext<TransactionContextType | undefined>(
  undefined,
);

export const useTransactions = () => useContext(TransactionContext);

export const TransactionProvider = ({
  children,
}: {
  children: React.ReactNode;
}) => {
  const [transactions, setTransactions] = useState<Transaction[]>([]);
  const [balance, setBalance] = useState<number>(1000);

  const addTransaction = (amount: string, account: Beneficiary) => {
    const newTransaction: Transaction = Transaction.instantiate({
      id: Date.now(),
      amount: parseFloat(amount),
      account,
    });
    setTransactions(prevTransactions => [...prevTransactions, newTransaction]);
    setBalance(prevBalance => prevBalance - parseFloat(amount));
  };

  return (
    <TransactionContext.Provider
      value={{transactions, addTransaction, balance}}>
      {children}
    </TransactionContext.Provider>
  );
};
