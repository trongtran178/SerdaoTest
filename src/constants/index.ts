export * from './regex';
export * from './cacheKeys';