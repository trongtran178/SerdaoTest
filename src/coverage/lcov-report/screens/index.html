
<!doctype html>
<html lang="en">

<head>
    <title>Code coverage report for screens</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <link rel="shortcut icon" type="image/x-icon" href="../favicon.png" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
    
<body>
<div class='wrapper'>
    <div class='pad1'>
        <h1><a href="../index.html">All files</a> screens</h1>
        <div class='clearfix'>
            
            <div class='fl pad1y space-right2'>
                <span class="strong">72.6% </span>
                <span class="quiet">Statements</span>
                <span class='fraction'>53/73</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">60% </span>
                <span class="quiet">Branches</span>
                <span class='fraction'>12/20</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">52.63% </span>
                <span class="quiet">Functions</span>
                <span class='fraction'>10/19</span>
            </div>
        
            
            <div class='fl pad1y space-right2'>
                <span class="strong">71.83% </span>
                <span class="quiet">Lines</span>
                <span class='fraction'>51/71</span>
            </div>
        
            
        </div>
        <p class="quiet">
            Press <em>n</em> or <em>j</em> to go to the next uncovered block, <em>b</em>, <em>p</em> or <em>k</em> for the previous block.
        </p>
        <template id="filterTemplate">
            <div class="quiet">
                Filter:
                <input type="search" id="fileSearch">
            </div>
        </template>
    </div>
    <div class='status-line medium'></div>
    <div class="pad1">
<table class="coverage-summary">
<thead>
<tr>
   <th data-col="file" data-fmt="html" data-html="true" class="file">File</th>
   <th data-col="pic" data-type="number" data-fmt="html" data-html="true" class="pic"></th>
   <th data-col="statements" data-type="number" data-fmt="pct" class="pct">Statements</th>
   <th data-col="statements_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="branches" data-type="number" data-fmt="pct" class="pct">Branches</th>
   <th data-col="branches_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="functions" data-type="number" data-fmt="pct" class="pct">Functions</th>
   <th data-col="functions_raw" data-type="number" data-fmt="html" class="abs"></th>
   <th data-col="lines" data-type="number" data-fmt="pct" class="pct">Lines</th>
   <th data-col="lines_raw" data-type="number" data-fmt="html" class="abs"></th>
</tr>
</thead>
<tbody><tr>
	<td class="file high" data-value="CreateBeneficiaryScreen.tsx"><a href="CreateBeneficiaryScreen.tsx.html">CreateBeneficiaryScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="4" class="abs high">4/4</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="17" class="abs high">17/17</td>
	</tr>

<tr>
	<td class="file high" data-value="HomeScreen.tsx"><a href="HomeScreen.tsx.html">HomeScreen.tsx</a></td>
	<td data-value="100" class="pic high">
	<div class="chart"><div class="cover-fill cover-full" style="width: 100%"></div><div class="cover-empty" style="width: 0%"></div></div>
	</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="2" class="abs high">2/2</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="3" class="abs high">3/3</td>
	<td data-value="100" class="pct high">100%</td>
	<td data-value="9" class="abs high">9/9</td>
	</tr>

<tr>
	<td class="file medium" data-value="TransactionScreen.tsx"><a href="TransactionScreen.tsx.html">TransactionScreen.tsx</a></td>
	<td data-value="57.44" class="pic medium">
	<div class="chart"><div class="cover-fill" style="width: 57%"></div><div class="cover-empty" style="width: 43%"></div></div>
	</td>
	<td data-value="57.44" class="pct medium">57.44%</td>
	<td data-value="47" class="abs medium">27/47</td>
	<td data-value="42.85" class="pct low">42.85%</td>
	<td data-value="14" class="abs low">6/14</td>
	<td data-value="35.71" class="pct low">35.71%</td>
	<td data-value="14" class="abs low">5/14</td>
	<td data-value="55.55" class="pct medium">55.55%</td>
	<td data-value="45" class="abs medium">25/45</td>
	</tr>

<tr>
	<td class="file empty" data-value="index.ts"><a href="index.ts.html">index.ts</a></td>
	<td data-value="0" class="pic empty">
	<div class="chart"><div class="cover-fill" style="width: 0%"></div><div class="cover-empty" style="width: 100%"></div></div>
	</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	<td data-value="0" class="pct empty">0%</td>
	<td data-value="0" class="abs empty">0/0</td>
	</tr>

</tbody>
</table>
</div>
                <div class='push'></div><!-- for sticky footer -->
            </div><!-- /wrapper -->
            <div class='footer quiet pad2 space-top1 center small'>
                Code coverage generated by
                <a href="https://istanbul.js.org/" target="_blank" rel="noopener noreferrer">istanbul</a>
                at 2024-08-19T17:00:47.575Z
            </div>
        <script src="../prettify.js"></script>
        <script>
            window.onload = function () {
                prettyPrint();
            };
        </script>
        <script src="../sorter.js"></script>
        <script src="../block-navigation.js"></script>
    </body>
</html>
    