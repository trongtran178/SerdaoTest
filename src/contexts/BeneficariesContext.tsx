import AsyncStorage from '@react-native-async-storage/async-storage';
import debounce from 'lodash/debounce';
import toString from 'lodash/toString';
import React, {
  createContext,
  useCallback,
  useContext,
  useEffect,
  useState,
} from 'react';
import {Beneficiary} from '../models';
import {notifyMessage} from '../utils';
import { ECacheKeys } from '../constants';

export type BeneficariesContextType = {
  beneficiaries: Beneficiary[];
  addBeneficiary: (beneficary: unknown) => Beneficiary | undefined;
};

const BeneficariesContext = createContext<BeneficariesContextType | undefined>(
  undefined,
);

export const useBeneficiaries = () => useContext(BeneficariesContext);

export const BeneficiariesProvider = ({
  children,
}: {
  children: React.ReactNode | React.ReactNode[];
}) => {
  const [beneficiaries, setBeneficiaries] = useState<Beneficiary[]>([]);

  useEffect(() => {}, []);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const saveDataToAsyncStorage = useCallback(
    debounce(async () => {
      try {
        await AsyncStorage.setItem(
          ECacheKeys.BENEFICIARIES_CONTEXT,
          JSON.stringify(beneficiaries),
        );
      } catch (e) { }
    }, 1000),
    [beneficiaries],
  );

  useEffect(() => {
    saveDataToAsyncStorage();
  }, [beneficiaries, saveDataToAsyncStorage]);

  useEffect(() => {
    const getBeneficiaries = async () => {
      const result = await AsyncStorage.getItem(ECacheKeys.BENEFICIARIES_CONTEXT);
      if (result) {
        const parsed = JSON.parse(result);
        setBeneficiaries(parsed);
      }
    };
    getBeneficiaries();
  }, []);

  const addBeneficiary = (beneficary: unknown) => {
    try {
      const newBeneficiary = Beneficiary.instantiate(beneficary);
      setBeneficiaries(prevBeneficiaries => [
        ...prevBeneficiaries,
        newBeneficiary,
      ]);
      notifyMessage('Beneficiary added successfully');
      return newBeneficiary;
    } catch (e) {
      notifyMessage(toString(e));
    }
  };
  return (
    <BeneficariesContext.Provider value={{beneficiaries, addBeneficiary}}>
      {children}
    </BeneficariesContext.Provider>
  );
};
