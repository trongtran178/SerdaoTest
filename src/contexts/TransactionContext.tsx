import React, {
  createContext,
  useState,
  useContext,
  useEffect,
  useCallback,
} from 'react';
import {Transaction, Account} from '../models';
import get from 'lodash/get';
import AsyncStorage from '@react-native-async-storage/async-storage';
import {debounce, isNumber} from 'lodash';
import { ECacheKeys } from '../constants';
export type TransactionContextType = {
  transactions: Transaction[];
  addTransaction: (amount: string, account: unknown) => void;
  balance: number;
};
const TransactionContext = createContext<TransactionContextType | undefined>(
  undefined,
);

export const useTransactions = () => useContext(TransactionContext);

export const TransactionProvider = ({
  children,
}: {
  children: React.ReactNode | React.ReactNode[];
}) => {
  const [transactions, setTransactions] = useState<Transaction[]>([]);
  const [balance, setBalance] = useState<number>(1000);

  const addTransaction = (amount: string, account: unknown) => {
    const accountTransaction = Account.instantiate({
      name: get(account, 'name', ''),
      iban: get(account, 'iban', ''),
    });
    const regex = /^-?\d+(\.\d+)?$/;
    
    if(/^-?\d+(\.\d+)?$/.test(amount) === false)
      throw new Error('Amount must be a number');
    const amountParsed = parseFloat(amount);
    const newTransaction: Transaction = Transaction.instantiate({
      id: Date.now(),
      amount: amountParsed,
      account: accountTransaction,
      currentBalance: balance,
    });
    setTransactions(prevTransactions => [...prevTransactions, newTransaction]);
    setBalance(prevBalance => prevBalance - parseFloat(amount));
  };

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const saveDataToAsyncStorage = useCallback(
    debounce(() => {
      try {
        AsyncStorage.setItem(
          ECacheKeys.TRANSACTIONS_CONTEXT,
          JSON.stringify({transactions, balance}),
        );
      } catch (e) {}
    }, 1000),
    [transactions, balance],
  );

  useEffect(() => {
    saveDataToAsyncStorage();
  }, [transactions, balance, saveDataToAsyncStorage]);

  useEffect(() => {
    const getTransactions = async () => {
      try {
        const data = await AsyncStorage.getItem(ECacheKeys.TRANSACTIONS_CONTEXT);
        if (data) {
          const {transactions: transactionsCached, balance: balanceCached} =
            JSON.parse(data);
          setTransactions(transactionsCached);
          setBalance(balanceCached);
        }
      } catch (e) {}
    };

    getTransactions();
  }, []);

  return (
    <TransactionContext.Provider
      value={{transactions, addTransaction, balance}}>
      {children}
    </TransactionContext.Provider>
  );
};
