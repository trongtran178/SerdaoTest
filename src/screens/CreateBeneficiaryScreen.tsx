import isNil from 'lodash/isNil';
import React, {useCallback, useState} from 'react';
import {Image, StyleSheet, Text, TouchableOpacity, View} from 'react-native';
import {CommonTextInput} from '../components/CommonTextInput';
import {useBeneficiaries} from '../contexts/BeneficariesContext';
import {ParamListBase} from '@react-navigation/native';
import {NativeStackScreenProps} from '@react-navigation/native-stack';
import {Icons} from '../assets';

type Props = NativeStackScreenProps<ParamListBase, 'CreateBeneficiary'>;

const CreateBeneficaryScreen = ({navigation}: Props) => {
  const [firstName, setFirstName] = useState<string>('');
  const [lastName, setLastName] = useState<string>('');
  const [iban, setIban] = useState<string>('');

  const beneficiariesContext = useBeneficiaries();

  if (isNil(beneficiariesContext)) {
    return null;
  }
  const {addBeneficiary} = beneficiariesContext;

  // eslint-disable-next-line react-hooks/rules-of-hooks
  const onCreate = useCallback(() => {
    const newBeneficiary = addBeneficiary({firstName, lastName, iban});
    if (!isNil(newBeneficiary)) {
      setFirstName('');
      setLastName('');
      setIban('');
      navigation.goBack();
    }
  }, [addBeneficiary, firstName, iban, lastName, navigation]);

  return (
    <View style={styles.container}>
      <View style={styles.inputContainer}>
        <CommonTextInput
          testID='textinput_firstname'
          onChangeText={setFirstName}
          value={firstName}
          maxLength={50}
          placeholder="Enter first name"
        />
        <View>
          <Image source={Icons.name} style={styles.icon} />
        </View>
      </View>
      <View style={styles.inputContainer}>
        <CommonTextInput
          testID='textinput_lastname'
          onChangeText={setLastName}
          value={lastName}
          maxLength={50}
          placeholder="Enter last name"
        />
        <View>
          <Image source={Icons.name} style={styles.icon} />
        </View>
      </View>

      <View style={styles.inputContainer}>
        <CommonTextInput
          testID='textinput_iban'
          onChangeText={setIban}
          value={iban}
          maxLength={34}
          placeholder="Enter IBAN number"
        />
        <View>
          <Image source={Icons.iban_number} style={styles.icon} />
        </View>
      </View>

      <TouchableOpacity style={styles.button} onPress={onCreate}>
        <Text style={styles.buttonText}>Create beneficiary</Text>
      </TouchableOpacity>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  textInput: {
    height: 40,
    borderColor: 'gray',
    borderWidth: 1,
    width: '80%',
    marginVertical: 8,
  },
  button: {
    backgroundColor: '#4CAF50', // Material Design green
    marginVertical: 12,
    padding: 12,
    borderRadius: 12,
    alignItems: 'center',
    elevation: 2, // for Android shadow
    shadowColor: '#000', // for iOS shadow
    shadowOffset: {width: 0, height: 2}, // for iOS shadow
    shadowOpacity: 0.1, // for iOS shadow
    shadowRadius: 5, // for iOS shadow
  },
  buttonText: {
    color: '#fff',
    fontSize: 16,
    fontWeight: 'bold',
  },
  icon: {
    position: 'absolute',
    top: -12,
    right: 12,
    width: 20,
    height: 20,
  },
  inputContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    borderRadius: 5,
  },
});

export default CreateBeneficaryScreen;
