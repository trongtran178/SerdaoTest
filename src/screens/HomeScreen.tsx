import React from 'react';
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  ScrollView,
} from 'react-native';
import {useTransactions} from '../contexts/TransactionContext';
import type {NativeStackScreenProps} from '@react-navigation/native-stack';
import isNil from 'lodash/isNil';
import {ParamListBase} from '@react-navigation/native';
import {CurrentBalanceCard} from '../components/CurrentBalanceCard';
import {TransactionList} from '../components/TransactionList';

export type Props = NativeStackScreenProps<ParamListBase, 'Home'>;

const HomeScreen = ({navigation}: Props) => {
  const transactionContext = useTransactions();
  if (isNil(transactionContext)) {
    return null;
  }
  const {transactions, balance} = transactionContext;

  return (
    <ScrollView style={styles.container}>
      <CurrentBalanceCard currentBalance={balance} />
      <View style={styles.actions}>
        <TouchableOpacity
          style={[styles.button, {marginRight: 4}, {backgroundColor: '#4CAF50'}]}
          onPress={() => navigation.navigate('Transaction')}>
          <Text style={[styles.buttonText]}>Add Transaction</Text>
        </TouchableOpacity>
        <TouchableOpacity
          style={[styles.button, {marginLeft: 4}]}
          onPress={() => navigation.navigate('CreateBeneficiary')}>
          <Text style={styles.buttonText}>Create Beneficiary</Text>
        </TouchableOpacity>
      </View>

      <Text style={styles.header}>Transaction History</Text>
      <TransactionList transactions={transactions} />
    </ScrollView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    margin: 12,
  },
  header: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333',
    marginTop: 8,
    marginBottom: 8,
  },
  item: {
    backgroundColor: '#f9f9f9',
    padding: 20,
    marginVertical: 8,
    marginHorizontal: 16,
    borderRadius: 5,
    borderWidth: 1,
    borderColor: '#ddd',
  },
  itemText: {
    fontSize: 16,
  },
  listContainer: {
    flexGrow: 1,
    width: '100%',
  },
  balanceText: {
    fontSize: 18,
    fontWeight: 'bold',
    color: '#333',
    marginBottom: 5,
  },
  balanceAmount: {
    fontSize: 24,
    fontWeight: 'bold',
    color: '#4CAF50', // Material Design green
  },
  actions: {
    flexDirection: 'row',
  },
  button: {
    flex: 0.5,
    marginVertical: 12,
    padding: 12,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#ddd',
    backgroundColor: '#2196F3', // Material Design blue
    alignItems: 'center',
    justifyContent: 'center',
  },
  buttonText: {
    fontSize: 16,
    color: '#fff',
    fontWeight: 'bold',
  },
});

export default HomeScreen;
