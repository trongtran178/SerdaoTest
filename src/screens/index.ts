import TransactionScreen from './TransactionScreen';
import HomeScreen from './HomeScreen';
import CreateBeneficiaryScreen from './CreateBeneficiaryScreen';
export {TransactionScreen, HomeScreen, CreateBeneficiaryScreen};
