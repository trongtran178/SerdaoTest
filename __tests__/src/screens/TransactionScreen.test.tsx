import { describe, it, expect, jest } from "@jest/globals";
import { ParamListBase } from "@react-navigation/native";
import { NativeStackNavigationProp } from "@react-navigation/native-stack";
import { render } from "@testing-library/react-native";
import { TransactionProvider } from "../../../src/contexts/TransactionContext";
import { TransactionScreen } from "../../../src/screens";
import { BeneficiariesProvider } from "../../../src/contexts/BeneficariesContext";

describe('TransactionScreen', () => {
  it('should render correctly', () => {
    const navigationMock = {
      navigate: jest.fn(),
      goBack: jest.fn(),
      canGoBack: jest.fn(),
      setOptions: jest.fn(),
      setParams: jest.fn(),
      dispatch: jest.fn(),
      isFocused: jest.fn(),
      dangerouslyGetState: jest.fn(),
      dangerouslyGetParent: jest.fn(),
      getState: jest.fn(),
      reset: jest.fn(),
      addListener: jest.fn(),
      removeListener: jest.fn(),
      addKeyedListener: jest.fn(),
      removeKeyedListener: jest.fn(),
      launch: jest.fn(),
      dismiss: jest.fn(),
      getParent: jest.fn(),
      getKey: jest.fn().mockReturnValue("Transaction"),
      isFirstRouteInParent: jest.fn().mockReturnValue(true),
      type: "native",
      name: "Transaction",
      replace: jest.fn(),
      push: jest.fn(),
      pop: jest.fn(),
      popToTop: jest.fn(),
    } as unknown as NativeStackNavigationProp<ParamListBase, "Transaction", undefined>;

    const routeMock = {
      key: "Transaction",
      name: "Transaction",
      params: {}
    } as const;

    const { toJSON } = render(
      <TransactionProvider>
        <BeneficiariesProvider>
            <TransactionScreen navigation={navigationMock} route={routeMock} />
        </BeneficiariesProvider>
      </TransactionProvider>
    )
    expect(toJSON()).toMatchSnapshot()
  })
})
