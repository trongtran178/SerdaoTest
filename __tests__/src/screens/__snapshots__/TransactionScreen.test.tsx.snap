// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TransactionScreen should render correctly 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      autoFocus={true}
      inputMode="decimal"
      keyboardType="numeric"
      onChangeText={[Function]}
      placeholder="Enter amount"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginBottom": 15,
          "marginVertical": 8,
          "paddingHorizontal": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
          "width": "80%",
        }
      }
      value=""
    />
    <View>
      <Image
        source={
          {
            "uri": "https://cdn-icons-png.flaticon.com/512/33/33007.png",
          }
        }
        style={
          {
            "height": 20,
            "position": "absolute",
            "right": 12,
            "top": -12,
            "width": 20,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Recipient Name"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginBottom": 15,
          "marginVertical": 8,
          "paddingHorizontal": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
          "width": "80%",
        }
      }
      value=""
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "borderRadius": 5,
        "flexDirection": "row",
      }
    }
  >
    <TextInput
      onChangeText={[Function]}
      placeholder="Recipient IBAN"
      style={
        {
          "backgroundColor": "#fff",
          "borderColor": "#ddd",
          "borderRadius": 10,
          "borderWidth": 1,
          "elevation": 2,
          "height": 50,
          "marginBottom": 15,
          "marginVertical": 8,
          "paddingHorizontal": 10,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 2,
            "width": 0,
          },
          "shadowOpacity": 0.1,
          "shadowRadius": 5,
          "width": "80%",
        }
      }
      value=""
    />
    <View>
      <Image
        source={
          {
            "uri": "https://cdn-icons-png.flaticon.com/512/4930/4930340.png",
          }
        }
        style={
          {
            "height": 20,
            "position": "absolute",
            "right": 12,
            "top": -12,
            "width": 20,
          }
        }
      />
    </View>
  </View>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "#4CAF50",
        "borderRadius": 12,
        "elevation": 2,
        "marginVertical": 12,
        "opacity": 1,
        "padding": 12,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 5,
      }
    }
  >
    <Text
      style={
        {
          "color": "#fff",
          "fontSize": 16,
          "fontWeight": "bold",
        }
      }
    >
      Submit Transaction
    </Text>
  </View>
  <View
    accessibilityHint="Tap to close the bottom sheet"
    accessibilityLabel="Bottom sheet backdrop"
    accessibilityRole="button"
    accessible={true}
    animatedPosition={1334}
    collapsable={false}
    forwardedRef={[Function]}
    handlerTag={1}
    handlerType="TapGestureHandler"
    jestAnimatedStyle={
      {
        "value": {
          "flex": 1,
          "opacity": 0,
        },
      }
    }
    onGestureHandlerEvent={[Function]}
    pointerEvents="auto"
    style={
      {
        "backgroundColor": "black",
        "bottom": 0,
        "flex": 1,
        "left": 0,
        "opacity": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    onLayout={[Function]}
    pointerEvents="box-none"
    style={
      [
        undefined,
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        },
        {
          "bottom": 0,
          "overflow": "hidden",
          "top": 0,
        },
      ]
    }
  >
    <View
      collapsable={false}
      jestAnimatedStyle={
        {
          "value": {
            "opacity": 1,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          },
        }
      }
      style={
        {
          "flexDirection": "column-reverse",
          "left": 0,
          "opacity": 1,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "transform": [
            {
              "translateY": 1334,
            },
          ],
        }
      }
    >
      <View
        accessibilityLabel="Bottom Sheet"
        accessibilityRole="adjustable"
        accessible={true}
        pointerEvents="none"
        style={
          [
            {
              "backgroundColor": "white",
              "borderRadius": 15,
            },
            {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
          ]
        }
      />
      <View
        accessibilityLabel="Bottom Sheet"
        accessibilityRole="adjustable"
        accessible={true}
        collapsable={false}
        jestAnimatedStyle={
          {
            "value": {
              "paddingBottom": -999,
            },
          }
        }
        pointerEvents="box-none"
        style={
          {
            "overflow": "hidden",
            "paddingBottom": -999,
          }
        }
      >
        <View
          collapsable={false}
          forwardedRef={[Function]}
          handlerTag={2}
          handlerType="PanGestureHandler"
          jestAnimatedStyle={
            {
              "value": {
                "height": 0,
              },
            }
          }
          onGestureHandlerEvent={[Function]}
          style={
            {
              "height": 0,
            }
          }
        >
          <View
            collapsable={false}
            forwardedRef={[Function]}
            handlerTag={3}
            handlerType="PanGestureHandler"
            jestAnimatedStyle={
              {
                "value": {},
              }
            }
            onGestureHandlerEvent={[Function]}
            style={
              {
                "flex": 1,
                "overflow": "visible",
              }
            }
          >
            <RCTScrollView
              ListHeaderComponent={<BottomSheetFlatListHeader />}
              collapsable={false}
              contentContainerStyle={
                {
                  "backgroundColor": "white",
                }
              }
              data={[]}
              decelerationRate={0}
              forwardedRef={[Function]}
              getItem={[Function]}
              getItemCount={[Function]}
              handlerTag={4}
              handlerType="NativeViewGestureHandler"
              jestAnimatedStyle={
                {
                  "value": {},
                }
              }
              keyExtractor={[Function]}
              keyboardDismissMode="interactive"
              onContentSizeChange={[Function]}
              onGestureHandlerEvent={[Function]}
              onLayout={[Function]}
              onMomentumScrollBegin={[Function]}
              onMomentumScrollEnd={[Function]}
              onScroll={[Function]}
              onScrollBeginDrag={[Function]}
              onScrollEndDrag={[Function]}
              overScrollMode="never"
              removeClippedSubviews={false}
              renderItem={[Function]}
              scrollEventThrottle={16}
              showsVerticalScrollIndicator={false}
              stickyHeaderIndices={[]}
              style={{}}
              viewabilityConfigCallbackPairs={[]}
            >
              <View>
                <View
                  collapsable={false}
                  onLayout={[Function]}
                >
                  <Text
                    style={
                      {
                        "alignSelf": "center",
                        "color": "#333",
                        "fontSize": 18,
                        "fontWeight": "bold",
                        "marginBottom": 8,
                        "marginTop": 8,
                        "textAlign": "center",
                      }
                    }
                  >
                    Choose Beneficiary
                  </Text>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        forwardedRef={[Function]}
        handlerTag={5}
        handlerType="PanGestureHandler"
        jestAnimatedStyle={
          {
            "value": {},
          }
        }
        onGestureHandlerEvent={[Function]}
        onLayout={[Function]}
      >
        <View
          accessibilityHint="Drag up or down to extend or minimize the bottom sheet"
          accessibilityLabel="Bottom sheet handle"
          accessibilityRole="adjustable"
          accessible={true}
          collapsable={false}
          jestAnimatedStyle={
            {
              "value": {},
            }
          }
          style={
            {
              "padding": 10,
            }
          }
        >
          <View
            collapsable={false}
            jestAnimatedStyle={
              {
                "value": {},
              }
            }
            style={
              {
                "alignSelf": "center",
                "backgroundColor": "rgba(0, 0, 0, 0.75)",
                "borderRadius": 4,
                "height": 4,
                "width": 56.25,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</View>
`;
